### PR TITLE
make billing_report.inc.php a class

### PR DESCRIPTION
**Pull request by stephenwaite**

### CODE REVIEW

**Comment by stephenwaite** [comment]
testing ok, for some reason had to hardcode the path to the class 
https://github.com/openemr/openemr/pull/2682/commits/781a1df4cc775faea6938e4c032ea72b1649a234#diff-fd6c028501caea9a8110d2af3252426fR179

File: interface/billing/sl_eob_invoice.php [hunk 1]
```diff
@@ -501,6 +501,12 @@ function updateFields(payField, adjField, balField, coPayField, isFirstProcCode)
                         <input type='text' name='form_stmt_count' id='form_stmt_count' class="form-control"
                                value='<?php echo attr((0 + $ferow['stmt_count'])); ?>'/>
                     </div>
+                    <div class="col-xs-2">
+                        <label class="control-label" for="form_last_bill"><?php echo xlt('Last Bill Date'); ?>
+                        :</label>
+                        <input type='text' name="form_last_bill" id='form_last_bill' class="form-control"
+                               value ='<?php echo attr($billdate); ?>' disabled/>
+                    </div>
```
Comment (bradymiller) [inline interface/billing/sl_eob_invoice.php]: This is odd since is already in the main codebase.

File: interface/billing/billing_report.php [hunk 2]
```diff
@@ -669,7 +668,7 @@ function divtoggle(spanid, divid) { //Called in the Expand, Collapse links(This
                 // The below section is needed if there is any 'include' type in the $ThisPageSearchCriteriaDataTypeMaster
                 // Function name is added here.Corresponding include files need to be included in the respective pages as done in this page.
                 // It is labled(Included for Insurance ajax criteria)(Line:-279-299).
-                $ThisPageSearchCriteriaIncludeMaster[1] = "InsuranceCompanyDisplay"; // This is php function defined in the file 'report.inc.php'
+                $ThisPageSearchCriteriaIncludeMaster[1] = "BillingReport::InsuranceCompanyDisplay"; // This is php function defined in OpenEMR\Billing\BillingReport
```
Comment (bradymiller) [inline interface/billing/billing_report.php]: Would do this instead:
```
$ThisPageSearchCriteriaIncludeMaster[1] = "OpenEMR\Billing\BillingReport::InsuranceCompanyDisplay";
```

File: interface/reports/criteria.tab.php [hunk 3]
```diff
@@ -172,7 +176,7 @@
                     <div class= "col-xs-12">
                         <label class="control-label" for=""><?php echo text($ThisPageSearchCriteriaDisplay[$ThisPageSearchCriteriaIndex]); ?>:</label>
                         <div <?php //Don't Use class =  'form-control'?>>
-                            <?php $FunctionName=$ThisPageSearchCriteriaIncludeMaster[$ThisPageSearchCriteriaInclude[$ThisPageSearchCriteriaIndex]];
+                            <?php $FunctionName = "OpenEMR\\Billing\\" . $ThisPageSearchCriteriaIncludeMaster[$ThisPageSearchCriteriaInclude[$ThisPageSearchCriteriaIndex]];
```
Comment (bradymiller) [inline interface/reports/criteria.tab.php]: Would revert above line change and make the recommended change in the above interface/billing/billing_report.php script

**Comment by bradymiller** [comment]
Regarding the hardcoding of path, see my comments above. This is because this code is running a function from a string, so forces us to include the full namepath:
https://www.designcise.com/web/tutorial/how-to-dynamically-invoke-a-class-method-in-php

File: interface/billing/sl_eob_invoice.php (hunk 1 above)
Comment (stephenwaite) [reply interface/billing/sl_eob_invoice.php]: woops, i branched off of that last commit which obviously caused this

Comment (stephenwaite) [reply interface/billing/sl_eob_invoice.php]: branched off last commit which in prob not kosher

File: interface/billing/billing_report.php (hunk 2 above)
Comment (stephenwaite) [reply interface/billing/billing_report.php]: ok

**Comment by stephenwaite** [comment]
made changes and testing is fine, guessing squash and merge will eliminate weirdness introduced by not branching directly off of master

**Comment by bradymiller** [comment]
hi @stephenwaite . Yep, squash and merge should work out just fine on this PR. Nice work!
crowd

**Comment by stephenwaite** [comment]
lol

**Pull request merged**